agent/node.go

```diff
@@ -622,15 +622,18 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 
 			select {
 			case <-ctx.Done():
+				m.Stop(context.Background()) // todo: this should be sync like other components
 			case <-n.waitRole(ctx, ca.AgentRole):
 			}
 
-			m.Stop(context.Background()) // todo: this should be sync like other components
 			<-done
 
 			ready = nil // ready event happens once, even on multiple starts
 			n.Lock()
 			n.manager = nil
+			if n.conn != nil {
+				n.conn.Close()
+			}
```

Comment:
Note: this is being added because otherwise the console scrolls errors like 

```
INFO[0037] grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial unix /tmp/swarm3.sock: connect: no such file or directory"; Reconnecting to "/tmp/swarm3.sock"
```